crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -108,6 +139,13 @@ pub struct Polyline2d<const N: usize> {
 }
 impl<const N: usize> Primitive2d for Polyline2d<N> {}
 
+impl<const N: usize> Polyline2d<N> {
+    /// Create a new `Polyline2d` from an array of vertices
+    pub fn new(vertices: [Vec2; N]) -> Self {
```

Comment:
Another trivial constructor, but we could instead make use of some traits to allow construction form an iterator?

Reply:
I'm not the best at iterators, but I tried to implement this change. I'm not exactly the most confident in it however so could you give it a check?

Reply:
The `FromIterator` impl is almost right :) And then in this method, you want `impl<IntoIterator<Vec2>> as your argument type.

Then, within the function body, you call `Self::from_iter(vertices.into_iter())` :)

Reply:
This should be implemented in my newest commit 👍